features/storage/FEATURE_STORAGE/cfstore/configuration-store/cfstore.h

```diff
@@ -0,0 +1,643 @@
+     *  cpp_thread1.cpp::cfstore_cpp_thread1_case_06a() test case code (error
+     *  handling removed).
+     *
+     * int32_t cfstore_cpp_thread1_case_06a(void)
+     * {
+     *     int32_t counter = 6;
+     *     size_t value_len = 0;
+     *     const char* key_name = "com.arm.mbed.count";
+     *     cfstore_keydesc_t kdesc;
+     *     Cfstore cfstore = Cfstore();
+     *     CfstoreKey *key = NULL;
+     *
+     *     cfstore.initialize();
+     *     memset(&kdesc, 0, sizeof(kdesc));
+     *     kdesc.drl = ARM_RETENTION_WHILE_DEVICE_ACTIVE;
+     *     value_len = sizeof(counter);
+     *     key = cfstore.create(key_name, value_len, &kdesc);
+     *     key->write((const char*) &counter, &value_len);
+     *     cfstore.flush();
+     *     key->close();
+     *     delete key;
+     *     cfstore.uninitialize();
+     *     return 0;
+     * }
+     */
+    int32_t write(const char *data, size_t *len);
+
+private:
+
+    /**
+     * @param   hkey
+     *  This is insecure storage in the caller box. Cfstore stores the pointer to
+     *  the cfstore secured storage in this buffer, which is opaque to the caller.
+     *  Should the caller dereference the pointer, uvisor which create a
+     *  protection fault.
+     */
+    uint8_t hkey[CFSTORE_HANDLE_BUFSIZE];   //!< Insecure key handle storage. Cfstore stores the pointer to
+                                            //!< the cfstore secured storage in this buffer, which is opaque to the caller.
+                                            //!< Should the caller dereference the opaque pointer, uvisor will generate a
+                                            //!< protection fault.
+    Cfstore *owner;                         //!< Pointer to cfstore that created this key.
+                                            //!< This is required so key can lock transactions.
+};
+
+/** @brief   Class representing interface to Configuration Store
+ */
+class Cfstore
+{
+    friend class CfstoreKey;
+public:
+    Cfstore(void);
+    ~Cfstore(void);
+
+    /** Create a key-value pair in the store.
+     *
+     * @param key_name      The name of the key.
+     * @param value_len     The size in bytes of the stored value.
+     * @param kdesc         The key descriptor specifying the properties of the key-value.
+     *
+     * @return
+     *  On success, the method returns a valid pointer to a CfstoreKey object through
+     *  which the KV can be manipulated. On failure, the method returns NULL;
+     *
+     * @note
+     *  The following example demonstrates the use of the create() method. The code
+     *  is a simplified version of the
+     *  cpp_thread1.cpp::cfstore_cpp_thread1_case_06a() test case code (error
+     *  handling removed).
+     *
+     * int32_t cfstore_cpp_thread1_case_06a(void)
+     * {
+     *     int32_t counter = 6;
+     *     size_t value_len = 0;
+     *     const char* key_name = "com.arm.mbed.count";
+     *     cfstore_keydesc_t kdesc;
```

Comment:
seems less than ideal that we have a structure, class for store and class for key. Given a key seems to be a filehandle did you look at the FileXXX classes for an existing pattern? It might fit. https://github.com/ARMmbed/mbed-os/tree/master/hal/api and look at all the files starting with File...